br/pkg/restore/client.go

```diff
@@ -2535,6 +2536,68 @@ func (rc *Client) UpdateSchemaVersion(ctx context.Context) error {
 	return nil
 }
 
+const (
+	alterTableDropIndexFormat      = "ALTER TABLE `%s`.`%s` DROP INDEX `%s`"
+	alterTableAddIndexFormat       = "ALTER TABLE `%s`.`%s` ADD INDEX `%s`(%s)"
+	alterTableAddUniqueIndexFormat = "ALTER TABLE `%s`.`%s` ADD UNIQUE KEY `%s`(%s)"
+	alterTableAddPrimaryFormat     = "ALTER TABLE `%s`.`%s` ADD PRIMARY KEY (%s) NONCLUSTERED"
+)
+
+// RepairIngestIndex drops the indexes from IngestRecorder and re-add them.
+func (rc *Client) RepairIngestIndex(ctx context.Context, ingestRecorder *ingestrec.IngestRecorder, g glue.Glue, storage kv.Storage) error {
+	dom, err := g.GetDomain(storage)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	info := dom.InfoSchema()
+	allSchema := info.AllSchemas()
+	ingestRecorder.UpdateIndexInfo(allSchema)
+	err = ingestRecorder.Iterate(func(tableID int64, _ int64, info *ingestrec.IngestIndexInfo) error {
+		var (
+			dropSQL      string = fmt.Sprintf(alterTableDropIndexFormat, info.SchemaName, info.TableName, info.IndexInfo.Name.O)
+			addSQL       string = ""
+			indexTypeSQL string = ""
+			commentSQL   string = ""
+			visibleSQL   string = " VISIBLE"
+		)
+
+		if info.IsPrimary {
+			addSQL = fmt.Sprintf(alterTableAddPrimaryFormat, info.SchemaName, info.TableName, info.ColumnList)
+		} else if info.IndexInfo.Unique {
+			addSQL = fmt.Sprintf(alterTableAddUniqueIndexFormat, info.SchemaName, info.TableName, info.IndexInfo.Name.O, info.ColumnList)
+		} else {
+			addSQL = fmt.Sprintf(alterTableAddIndexFormat, info.SchemaName, info.TableName, info.IndexInfo.Name.O, info.ColumnList)
+		}
+		// USING BTREE/HASH/RTREE
+		indexTypeStr := info.IndexInfo.Tp.String()
+		if len(indexTypeStr) > 0 {
+			indexTypeSQL = fmt.Sprintf(" USING %s", indexTypeStr)
+		}
+
+		// COMMENT [...]
+		if len(info.IndexInfo.Comment) > 0 {
+			commentSQL = " COMMENT %?"
+		}
+
+		if info.IndexInfo.Invisible {
+			visibleSQL = " INVISIBLE"
+		}
+
+		addSQL = fmt.Sprintf("%s%s%s%s", addSQL, indexTypeSQL, commentSQL, visibleSQL)
+
+		log.Debug("repair ingest sql", zap.String("drop", dropSQL))
+		if err := rc.db.se.Execute(ctx, dropSQL); err != nil {
+			return errors.Trace(err)
+		}
+		log.Debug("repair ingest sql", zap.String("add", addSQL))
+		if err := rc.db.se.ExecuteInternal(ctx, addSQL, info.IndexInfo.Comment); err != nil {
+			return errors.Trace(err)
+		}
```

Comment:
Because the `Comment` has some escape characters such as `"` and `\n`, so here use `ExecuteInter` to send with args instead of plaintext.
The function pattern is as follow:
`Execute(context, sql string)`
`ExecuteInternal(context, sql string, args...)`

Reply:
done, both use `ExecuteInternal`.